oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2AutoConfig.java

```diff
@@ -65,4 +72,26 @@ OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler(OktaOAuth2Prope
     OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService(Collection<AuthoritiesProvider> authoritiesProviders) {
         return new OktaOidcUserService(oAuth2UserService(authoritiesProviders), authoritiesProviders);
     }
+
+    @Bean
+    public JwtAuthenticationConverter jwtAuthenticationConverter(OktaOAuth2Properties oktaOAuth2Properties) {
+        OktaJwtAuthenticationConverter converter = new OktaJwtAuthenticationConverter(oktaOAuth2Properties.getGroupsClaim());
+        converter.setJwtGrantedAuthoritiesConverter(new JwtGrantedAuthoritiesConverter());
+        return converter;
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    @ConditionalOnDefaultWebSecurity
+    static class OAuth2SecurityFilterChainConfiguration {
+        @Bean
+        SecurityFilterChain oauth2SecurityFilterChain(HttpSecurity http) throws Exception {
+            // as of Spring Security 5.4 the default chain uses oauth2Login OR a JWT resource server (NOT both)
+            // this does the same as both defaults merged together (and provides the previous behavior)
+            http.authorizeRequests((requests) -> requests.anyRequest().authenticated());
+            http.oauth2Login(Customizer.withDefaults());
+            http.oauth2Client();
+            http.oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
```

Comment:
Nice!

Reply:
In [this example](https://github.com/spring-projects/spring-security/blob/master/samples/boot/oauth2webclient-webflux/src/main/java/sample/config/SecurityConfig.java), I noticed they use `withDefaults()` for login, client, and resource server. Not sure what that does, but thought I'd mention it.

Reply:
I just checked `withDefaults()`, just adds a no-op "customizer"